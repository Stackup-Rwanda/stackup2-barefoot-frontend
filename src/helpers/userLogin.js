import dotenv from 'dotenv';

dotenv.config();
const baseUrl = process.env.BASE_URL;
const login = async (user) => {
  let result = '';
  const url = `${baseUrl}/auth/login`;
  result = await fetch(url, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(user),
  })
    .then((response) => response.json())
    .then((data) => data)
    .catch((error) => error);
  return result;
};
export default login;
