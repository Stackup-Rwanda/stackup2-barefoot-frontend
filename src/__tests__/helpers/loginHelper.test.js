/* eslint-disable no-undef */
import 'regenerator-runtime/runtime';
import userLogin from '../../helpers/userLogin';
import global from '../action/loginAction.test';

describe('login helpers', () => {
  it('login functionality success', async () => {
    const credentials = {
      email: 'travel@gmail.com',
      password: 'TravelAdmin2@',
    };
    await userLogin(credentials);
    expect(fetch).toHaveBeenCalledTimes(1);
  });
  it('login functionality fail', async () => {
    const credentials = {
      email: 'travel',
      password: 'Travel',
    };
    global.fetch = jest.fn(() => Promise.reject());
    await userLogin(credentials);
    expect(fetch).toHaveBeenCalledTimes(1);
  });
});
