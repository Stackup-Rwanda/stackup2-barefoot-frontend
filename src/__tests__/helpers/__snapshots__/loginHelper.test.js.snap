// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login actions Dispatches the correct action and payload 1`] = `Array []`;

exports[`login actions Dispatches the correct action and payload of wrong email 1`] = `
Array [
  Object {
    "payload": Object {
      "error": "Invalid email",
    },
    "type": "LOGIN_HANDLE",
  },
]
`;
